### PR TITLE
test: use a local sinopia server for e2e testing

### LINKED ISSUE #56: Benchmark suite
I would be up for working on this.

My thinking is that I will mock the network endpoints (registry mocker here: https://github.com/npm/npm-registry-mock) so that it's not network dependent. Could also be integrated with Travis, although I'm guessing that the performance might differ greatly based upon the VM you get assigned when the build is run.

@rstacruz: Do you have any thoughts?

### PR DESCRIPTION
**Pull request by zkochan**
Close #56

### CODE REVIEW

**Comment by zkochan** [comment]
This was needed because we need the possibility of changing dist-tags of packages from tests in order to write test for [pnpm update](https://github.com/rstacruz/pnpm/issues/300)

**Comment by iamstarkov** [comment]
will review tomorrow in the morning

File: package.json [hunk 1]
```diff
@@ -105,7 +107,9 @@
     "commit": "git-cz",
     "cover": "istanbul cover test/index.js",
     "lint": "standard && tslint -c tslint.json ./src/**/*.ts ./test/**/*.ts",
-    "test": "npm run lint && ts-node test | tap-diff",
+    "sinopia": "sinopia -c ./sinopia/config.yaml",
+    "test:e2e": "ts-node test | tap-diff",
+    "test": "npm run lint && npm-run-all -p -r sinopia test:e2e",
```
Comment (iamstarkov) [inline package.json]: "test": "npm-run-all lint -p -r sinopia test:e2e",
regarding docs https://github.com/mysticatea/npm-run-all/blob/master/docs/npm-run-all.md#run-a-mix-of-sequential-and-parallel-scripts

Comment (zkochan) [reply package.json]: thanks! done

**Pull request merged**